src/fixtures.ts

```diff
@@ -19,26 +19,55 @@ import makeDir from 'make-dir';
 import * as path from 'path';
 import * as tmp from 'tmp';
 
+export class FixtureContent {
+  constructor(public content: string | Fixtures, public mode = 0o000) {}
+  toFixture(key: string): Fixtures {
+    return { [key]: this.content };
+  }
+}
+
 export interface Fixtures {
   // If string, we create a file with that string contents. If fixture, we
   // create a subdirectory and recursively install the fixture.
   // TODO: support buffers to allow non-text files.
-  [name: string]: string | Fixtures;
+  [name: string]: string | Fixtures | FixtureContent;
 }
 
-export async function setupFixtures(dir: string, fixtures: Fixtures) {
-  await makeDir(dir);
+export async function setupFixtures(
+  dir: string,
+  fixtures: Fixtures
+): Promise<string[]> {
+  let inaccessibleFixtures: string[] = [];
   const keys = Object.keys(fixtures);
+
   for (const key of keys) {
     const filePath = path.join(dir, key);
-    if (typeof fixtures[key] === 'string') {
-      const contents = fixtures[key] as string;
-      await fs.writeFileSync(filePath, contents);
+    const contents = fixtures[key];
+
+    if (typeof contents === 'string') {
+      fs.writeFileSync(filePath, contents);
+    } else if (contents instanceof FixtureContent) {
```

Comment:
I wonder if it would be preferable to use looser structural typing here rather than nominal. My recommendation would be to define `FixtureContent` with:

```ts
interface FixtureWithMode {
  content: string | Fixture;
  mode: number;
};

function isFixtureWithMode(fix: Fixtures) : fix is FixtureWithMode {
  return 'number' === typeof (fix as FixtureWithMode).mode;
} 
```

This has the benefit that the users don't have to instantiate `FixtureContent` objects, and simply use object notation.

Reply:
Actually I already tried this solution (see the other [branch](https://github.com/pliski/inline-fixtures/blob/feat/inaccessible-with-interface/src/fixtures.ts) ) but then I was seduced by the toFixture() method. ... rolling back to Interface branch ...